specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-11-09/kusto.json

```diff
@@ -0,0 +1,3986 @@
+          }
+        },
+        "description": "Gets a Kusto cluster database principalAssignment.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ClusterNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DatabaseNameParameter"
+          },
+          {
+            "$ref": "#/parameters/PrincipalAssignmentNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The Kusto cluster database principal assignment object.",
+            "schema": {
+              "$ref": "#/definitions/DatabasePrincipalAssignment"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "DatabasePrincipalAssignments"
+        ],
+        "operationId": "DatabasePrincipalAssignments_CreateOrUpdate",
+        "x-ms-examples": {
+          "KustoDatabasePrincipalAssignmentsCreateOrUpdate": {
+            "$ref": "./examples/KustoDatabasePrincipalAssignmentsCreateOrUpdate.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "description": "Creates a Kusto cluster database principalAssignment.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ClusterNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DatabaseNameParameter"
+          },
+          {
+            "$ref": "#/parameters/PrincipalAssignmentNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/DatabasePrincipalAssignment"
+            },
+            "description": "The Kusto principalAssignments parameters supplied for the operation."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully updated the PrincipalAssignments.",
+            "schema": {
+              "$ref": "#/definitions/DatabasePrincipalAssignment"
+            }
+          },
+          "201": {
+            "description": "Successfully created the principalAssignments.",
+            "schema": {
+              "$ref": "#/definitions/DatabasePrincipalAssignment"
+            }
```

Comment:
Can this return 202 for async response?

Reply:
put doesn't return 202 for async response. Please check Clusters_CreateOrUpdate